app/helpers/textual_summary_helper.rb

```diff
@@ -1,3 +1,171 @@
+##
+# Base class for the textual summary DSL contexts.
+class BaseContext
```

Comment:
This is very generically named for a top-level class.  I suggest wrapping all of these classes under a namspace...perhaps...

```ruby
module TextualSummary
  class BaseContext
  class Context < BaseContext
  class BigGroupContext < Context
  ...
```

Reply:
A separate module seems good.  I generally do that when I have some implementation separate from what I want to mixin.

Additionally you can declare them as "private api" in a simple comment.  It doesn't enforce the privacy, but at least the caller can understand your intent.

I did that here for example: https://github.com/ManageIQ/more_core_extensions/blob/master/lib/more_core_extensions/core_ext/string/hex_dump.rb#L16